README.md
[![Actions Status](https://github.com/sanko/Alien-SDL2_image/workflows/CI/badge.svg)](https://github.com/sanko/Alien-SDL2_image/actions)
[![GitHub tag](https://img.shields.io/github/tag/sanko/Alien-SDL2_image.svg)]()
[![Cpan version](https://img.shields.io/cpan/v/Alien-SDL2_image.svg)](https://metacpan.org/release/Alien-SDL2_image)

# NAME

Alien::sdl\_image - Find or download and install SDL\_image

# SYNOPSIS

```perl
use strict;
use warnings;
use ExtUtils::MakeMaker;
use Config;
use Alien::Base::Wrapper ();
WriteMakefile(
  Alien::Base::Wrapper->new('Alien::libsdl2')->mm_args2(
    ...
    CONFIGURE_REQUIRES => {
      'Alien::libsdl2' => '0'
    },
    ...
  ),
);
```

# DESCRIPTION

This package can be used by other [CPAN](https://metacpan.org) modules that
require [libsdl2](http://libsdl.org).

# SEE ALSO

- [libsdl2](http://libsdl.org)
- [Alien::SDL2](https://metacpan.org/pod/Alien%3A%3ASDL2) - Ancient Alien for SDL2 (builds w/ SDL2-2.0.0 from 2013)

# Platform Support

This dist has been tested on various modern versions of perl on Windows Server
2019, Ubuntu (20.04.2), and OSX (10.15.7). See
https://github.com/sanko/Alien-libsdl2/actions

# LICENSE

Copyright (C) Sanko Robinson.

This library is free software; you can redistribute it and/or modify it under
the terms found in the Artistic License 2. Other copyrights, terms, and
conditions may apply to data transmitted through this module.

# AUTHOR

Sanko Robinson <sanko@cpan.org>


Changes
Revision history for Alien::libsdl2

1.08

	- Let's find out!

1.07

	- 1.06 but for real this time on Win32

1.06

	- Force share install (we need headers for SDL2::FFI & fixes ffi bundle on Win32)

1.05

	- Lib order for ffi bundle is broken (and I'm okay with that)

1.04

	- Attempt to build lib on Win32 for shared install

1.03

	- Prep for bundle

1.02

	- Attempt to build libs on bsd

1.01

	- Attempt to build libs on darwin/OSX

1.0       2021-05-17 20:10:38-04:00 America/New_York

	- Alien::SDL2 has been stalled for nearly a decade and I feel like making some games


lib/Alien/SDL2_image.pm
package Alien::SDL2_image;
use strict;
use warnings;
use base qw[Alien::Base];
our $VERSION = '1.08';

1;
__END__

=encoding utf8

=head1 NAME

Alien::sdl_image - Find or download and install SDL_image

=head1 SYNOPSIS

    use strict;
    use warnings;
    use ExtUtils::MakeMaker;
    use Config;
    use Alien::Base::Wrapper ();
    WriteMakefile(
      Alien::Base::Wrapper->new('Alien::libsdl2')->mm_args2(
        ...
        CONFIGURE_REQUIRES => {
          'Alien::libsdl2' => '0'
        },
        ...
      ),
    );

=head1 DESCRIPTION

This package can be used by other L<CPAN|https://metacpan.org> modules that
require L<libsdl2|http://libsdl.org>.

=head1 SEE ALSO

=over

=item L<libsdl2|http://libsdl.org>

=item L<Alien::SDL2> - Ancient Alien for SDL2 (builds w/ SDL2-2.0.0 from 2013)

=back

=head1 Platform Support

This dist has been tested on various modern versions of perl on Windows Server
2019, Ubuntu (20.04.2), and OSX (10.15.7). See
https://github.com/sanko/Alien-libsdl2/actions

=head1 LICENSE

Copyright (C) Sanko Robinson.

This library is free software; you can redistribute it and/or modify it under
the terms found in the Artistic License 2. Other copyrights, terms, and
conditions may apply to data transmitted through this module.

=head1 AUTHOR

Sanko Robinson E<lt>sanko@cpan.orgE<gt>

=cut


Makefile.PL
# This Makefile.PL for Alien-SDL2_image was generated by
# Dist::Zilla::Plugin::MakeMaker::Awesome 0.48.
# Don't edit it but the dist.ini and plugins used to construct it.

use strict;
use warnings;

use 5.008001;
use ExtUtils::MakeMaker 7.12;

my %WriteMakefileArgs = (
  "ABSTRACT" => "Find or download and install SDL_image",
  "AUTHOR" => "Sanko Robinson <sanko\@cpan.org>",
  "BUILD_REQUIRES" => {
    "Alien::Build" => "2.40",
    "Alien::Build::MM" => "0.32",
    "Alien::Build::Plugin::Build::Autoconf" => "2.40",
    "Alien::Build::Plugin::Build::Make" => "2.40",
    "Alien::gmake" => 0,
    "Alien::libsdl2" => "== 1.06",
    "Config" => 0,
    "ExtUtils::MakeMaker" => "7.1101",
    "IPC::Cmd" => 0,
    "Sort::Versions" => 0
  },
  "CONFIGURE_REQUIRES" => {
    "Alien::Build" => "2.40",
    "Alien::Build::MM" => "2.40",
    "Alien::Build::Plugin::Build::Make" => "2.40",
    "Alien::libsdl2" => "== 1.06",
    "ExtUtils::MakeMaker" => "7.12"
  },
  "DISTNAME" => "Alien-SDL2_image",
  "LICENSE" => "artistic_2",
  "MIN_PERL_VERSION" => "5.008001",
  "NAME" => "Alien::SDL2_image",
  "PREREQ_PM" => {
    "Alien::Base" => "2.40",
    "ExtUtils::MakeMaker" => "7.1101",
    "base" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "ExtUtils::MakeMaker" => "7.1101",
    "FFI::Platypus" => 0,
    "File::Spec" => 0,
    "Test2::Suite" => 0,
    "Test2::V0" => 0,
    "Test::Alien" => 0,
    "Test::More" => "0.88"
  },
  "VERSION" => "1.08",
  "test" => {
    "TESTS" => "t/*.t"
  }
);

my %FallbackPrereqs = (
  "Alien::Base" => "2.40",
  "Alien::Build" => "2.40",
  "Alien::Build::MM" => "0.32",
  "Alien::Build::Plugin::Build::Autoconf" => "2.40",
  "Alien::Build::Plugin::Build::Make" => "2.40",
  "Alien::gmake" => 0,
  "Alien::libsdl2" => "== 1.06",
  "Config" => 0,
  "ExtUtils::MakeMaker" => "7.1101",
  "FFI::Platypus" => 0,
  "File::Spec" => 0,
  "IPC::Cmd" => 0,
  "Sort::Versions" => 0,
  "Test2::Suite" => 0,
  "Test2::V0" => 0,
  "Test::Alien" => 0,
  "Test::More" => "0.88",
  "base" => 0,
  "strict" => 0,
  "warnings" => 0
);

# BEGIN code inserted by Dist::Zilla::Plugin::AlienBuild
use Alien::Build::MM;
my $abmm = Alien::Build::MM->new;
%WriteMakefileArgs = $abmm->mm_args(%WriteMakefileArgs);
# END code inserted by Dist::Zilla::Plugin::AlienBuild

unless ( eval { ExtUtils::MakeMaker->VERSION('6.63_03') } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);

# BEGIN code inserted by Dist::Zilla::Plugin::AlienBuild
{ package
    MY;
  sub postamble {
    $abmm->mm_postamble(@_);
  }
  sub install {
    $abmm->can('mm_install')
      ? $abmm->mm_install(@_)
      : shift->SUPER::install(@_);
  }
}
# END code inserted by Dist::Zilla::Plugin::AlienBuild
